docs/rules/prefer-array-flat.md

```diff
@@ -0,0 +1,53 @@
+# Prefer `Array#flat()` over legacy techniques to flatten arrays
+
+ES2019 introduced a new method [`Array#flat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) that flattens arrays, and there's a [`depth` parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat#parameters), so you can flatten deep nesting arrays with it.
```

Comment:
I don't think the `depth` parameter is relevant. People can read about it when they click the MDN link.